yt_dlp/extractor/nytimes.py

```diff
@@ -124,139 +88,348 @@ class NYTimesIE(NYTimesBaseIE):
-            'title': 'The Rise of Hitler',
-            'ext': 'mp3',
-            'description': 'md5:bce877fd9e3444990cb141875fab0028',
-            'creator': 'Pamela Paul',
-            'duration': 3475,
+            'id': '100000009202270',
+            'ext': 'mp4',
+            'title': 'Kamala Harris Defends Biden Policies, but Says ‘More Work’ Needed to Reach Voters',
+            'description': 'md5:de4212a7e19bb89e4fb14210ca915f1f',
+            'timestamp': 1701290997,
+            'upload_date': '20231129',
+            'uploader': 'By The New York Times',
+            'creator': 'Katie Rogers',
+            'thumbnail': r're:https?://\w+\.nyt.com/images/.*\.jpg',
+            'duration': 97631,
         },
         'params': {
+            # inconsistant md5
             'skip_download': True,
         },
     }, {
-        'url': 'http://www.nytimes.com/news/minute/2014/03/17/times-minute-whats-next-in-crimea/?_php=true&_type=blogs&_php=true&_type=blogs&_r=1',
+        # multiple videos in the same article
+        'url': 'https://www.nytimes.com/2023/12/02/business/air-traffic-controllers-safety.html',
+        'info_dict': {
+            'id': 'air-traffic-controllers-safety',
+            'title': 'Drunk and Asleep on the Job: Air Traffic Controllers Pushed to the Brink',
+            'description': 'md5:549e5a5e935bf7d048be53ba3d2c863d',
+            'upload_date': '20231202',
+            'creator': 'Emily Steel, Sydney Ember',
+            'timestamp': 1701511264,
+        },
+        'playlist_count': 3,
+
+    }, {
+        'url': 'https://www.nytimes.com/2023/12/02/business/media/netflix-squid-game-challenge.html',
         'only_matching': True,
     }]
 
-    def _extract_podcast_from_json(self, json, page_id, webpage):
-        podcast_audio = self._parse_json(
-            json, page_id, transform_source=js_to_json)
+    def _extract_content_from_block(self, block):
+        details = traverse_obj(block, {
+            'id': ('sourceId', {str}),
+            'uploader': ('bylines', ..., 'renderedRepresentation', {str}),
+            'duration': (('duration', 'length', ...), {int_or_none}),
+            'timestamp': ('firstPublished', {parse_iso8601}),
+            'series': ('podcastSeries', {str}),
+        }, get_all=False)
+
+        formats, subtitles = self._extract_media_from_json(details.get('id'), block)
 
-        audio_data = podcast_audio['data']
-        track = audio_data['track']
+        # audio articles will have an url and no formats
+        url = traverse_obj(block, ('fileUrl', {url_or_none}))
+        if not formats and url:
+            formats.append({'url': url, 'vcodec': 'none'})
 
-        episode_title = track['title']
-        video_url = track['source']
+        return {
+            **details,
+            'formats': formats,
+            'thumbnails': self._extract_thumbnails(traverse_obj(
+                block, ('promotionalMedia', 'crops', ..., 'renditions', ...))),
```

Comment:
nitpick: it's weird having thumbnails sandwiched between formats and subtitles
```suggestion
            'thumbnails': self._extract_thumbnails(traverse_obj(
                block, ('promotionalMedia', 'crops', ..., 'renditions', ...))),
            'formats': formats,
```